lib/YAML/Tiny.pm

```diff
@@ -528,7 +528,7 @@ sub _load_hash {
                 $hash->{$key} = undef;
                 return 1;
             }
-            if ( $lines->[0] =~ /^(\s*)-/ ) {
+            if ( $lines->[0] =~ /^(\s*)-(?:\s|$)/ ) {
```

Comment:
White space as defined in the YAML space is stricter than Perl's `\s`: YAML white space is only spaces and tab
http://www.yaml.org/spec/1.2/spec.html#id2775170

Reply:
I had completely forgotten about this PR. Sorry about that.

I've updated this PR and rebased it on `develop` branch